bgpd/bgp_nht.c

```diff
@@ -404,13 +421,42 @@ int bgp_find_or_add_nexthop(struct bgp *bgp_route, struct bgp *bgp_nexthop,
 				   peer);
 	} else {
 		if (BGP_DEBUG(nht, NHT))
-			zlog_debug("Found existing bnc %pFX(%d)(%s) flags 0x%x ifindex %d #paths %d peer %p, resolved prefix %pFX",
-				   &bnc->prefix, bnc->ifindex_ipv6_ll,
+			zlog_debug("Found existing bnc %pFX(%d)(%u)(%s) flags 0x%x ifindex %d #paths %d peer %p, resolved prefix %pFX",
+				   &bnc->prefix, bnc->ifindex_ipv6_ll, bnc->srte_color,
 				   bnc->bgp->name_pretty, bnc->flags,
 				   bnc->ifindex_ipv6_ll, bnc->path_count,
 				   bnc->nht_info, &bnc->resolved_prefix);
 	}
 
+	if (srte_color) {
+		if (make_prefix(afi, pi, &p, true) < 0)
```

Comment:
1) what about MPLS ? that means srte mpls will bring a new context that we don't need. can we protect it ?

2) this design puts the intelligence in BGP rather than in ZEBRA. before, MPLS TE was handled in ZEBRA, and now we decide to use bgp nht for srv6.
not sure I adhere to this approach. (I would need more to know about the advantages of choosing this).

in the separate pull request , we added nexthop tracking to pathd, and in that way, we don't need to modify BGP.